lte/gateway/c/core/oai/tasks/amf/amf_recv.cpp

```diff
@@ -207,6 +208,44 @@ int amf_handle_registration_request(
       sizeof(tai_t));
 
   if (msg->m5gs_reg_type.type_val == AMF_REGISTRATION_TYPE_INITIAL) {
+    OAILOG_INFO(LOG_NAS_AMF, "New REGITRATION_REQUEST processing\n");
+    // Check integrity and ciphering algorithm bits
+    // If all bits are zero it means integrity and ciphering algorithms are not
+    // valid, AMF should reject the initial registration. Note : amf_cause is
+    // upto network provider for invalid algorithms, here we considering
+    // CONDITIONAL_IE_ERROR as amf cause.
+    if (ue_context->amf_context.ue_sec_capability.ia == 0 ||
+        ue_context->amf_context.ue_sec_capability.ea == 0) {
+      OAILOG_ERROR(LOG_NAS_AMF, "UE is not supporting any algorithms");
+      amf_cause = AMF_CAUSE_UE_SEC_CAP_MISSMATCH;
+      OAILOG_ERROR(
+          LOG_NAS_AMF, "AMF rejecting the initial registration with cause %d",
+          amf_cause);
```

Comment:
can we merge 2 log in one ?
Generally could we agree on prefixing when feasible traces with UE identity (IMSI or ngap ids)

Reply:
Fixed

Reply:
fixed